incubator/hnc/internal/reconcilers/setup.go

```diff
@@ -9,25 +9,51 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
 	"sigs.k8s.io/controller-runtime/pkg/event"
 
 	api "sigs.k8s.io/multi-tenancy/incubator/hnc/api/v1alpha2"
 	"sigs.k8s.io/multi-tenancy/incubator/hnc/internal/forest"
 )
 
+type FakeDeleteCRDClient struct{}
+
+func (f FakeDeleteCRDClient) Get(context.Context, types.NamespacedName, runtime.Object) error {
+	return nil
+}
+
 var crds = map[string]bool{
 	"hncconfigurations.hnc.x-k8s.io":       true,
 	"subnamespaceanchors.hnc.x-k8s.io":     true,
 	"hierarchyconfigurations.hnc.x-k8s.io": true,
 }
 
+var deleteCRDClient interface {
```

Comment:
I had to think about this a bit - it might be nice to define this in two parts, the interface and the variable.

```
type interface deleteCRDClientType { ... }
var deleteCRDClient deleteCRDClientType
```

with a bit more commenting for both of them.